pylot/perception/detection/detection_operator.py

```diff
@@ -40,42 +40,20 @@ def __init__(self, camera_stream: erdos.ReadStream,
         self._logger = erdos.utils.setup_logging(self.config.name,
                                                  self.config.log_file_name)
         self._obstacles_stream = obstacles_stream
-        self._detection_graph = tf.Graph()
-        # Load the model from the model file.
-        pylot.utils.set_tf_loglevel(logging.ERROR)
-        with self._detection_graph.as_default():
-            od_graph_def = tf.GraphDef()
-            with tf.gfile.GFile(model_path, 'rb') as fid:
-                serialized_graph = fid.read()
-                od_graph_def.ParseFromString(serialized_graph)
-                tf.import_graph_def(od_graph_def, name='')
-
-        self._gpu_options = tf.GPUOptions(
-            allow_growth=True,
-            visible_device_list=str(self._flags.obstacle_detection_gpu_index),
-            per_process_gpu_memory_fraction=flags.
-            obstacle_detection_gpu_memory_fraction)
-        # Create a TensorFlow session.
-        self._tf_session = tf.Session(
-            graph=self._detection_graph,
-            config=tf.ConfigProto(gpu_options=self._gpu_options))
-        # Get the tensors we're interested in.
-        self._image_tensor = self._detection_graph.get_tensor_by_name(
-            'image_tensor:0')
-        self._detection_boxes = self._detection_graph.get_tensor_by_name(
-            'detection_boxes:0')
-        self._detection_scores = self._detection_graph.get_tensor_by_name(
-            'detection_scores:0')
-        self._detection_classes = self._detection_graph.get_tensor_by_name(
-            'detection_classes:0')
-        self._num_detections = self._detection_graph.get_tensor_by_name(
-            'num_detections:0')
+
+        physical_devices = tf.config.experimental.list_physical_devices('GPU') 
+        # Equivalent to allow_growth=True
+        for device in physical_devices:
+            tf.config.experimental.set_memory_growth(device, True)
+
+        # Load the model from the saved_model format file.
+        self._model = tf.saved_model.load('pylot/perception/detection/converted_model/')
+
         self._coco_labels = load_coco_labels(self._flags.path_coco_labels)
         self._bbox_colors = load_coco_bbox_colors(self._coco_labels)
         # Unique bounding box id. Incremented for each bounding box.
         self._unique_id = 0
-        # Serve some junk image to load up the model.
-        self.__run_model(np.zeros((108, 192, 3)))
+
```

Comment:
Why was the call to `__run_model` removed?

Reply:
Added it back -- I'm not entirely sure of it's purpose however

Reply:
I think Tensorflow 1.x lazily loads the model so we would usually incur an extremely high latency for perception of the first image. To prevent that, we run the model with a junk image to load the model into the memory at the operator construction time.